src/java.base/share/classes/java/lang/String.java

```diff
@@ -874,21 +866,33 @@ private static Charset lookupCharset(String csn) throws UnsupportedEncodingExcep
 
     private static byte[] encode(Charset cs, byte coder, byte[] val) {
         if (cs == UTF_8.INSTANCE) {
-            return encodeUTF8(coder, val, true);
+            return encodeUTF8(coder, val);
         }
         if (cs == ISO_8859_1.INSTANCE) {
             return encode8859_1(coder, val);
         }
         if (cs == US_ASCII.INSTANCE) {
             return encodeASCII(coder, val);
         }
-        return encodeWithEncoder(cs, coder, val, true);
+        return encodeWithEncoder(cs, coder, val);
+    }
+
+    private static byte[] encodeWithEncoder(Charset cs, byte coder, byte[] val) {
+        return encodeWithEncoder(cs, coder, val, null);
+    }
+
+    private static byte[] encodeWithEncoderNoReplacement(Charset cs, byte coder, byte[] val)
+            throws CharacterCodingException {
+        return encodeWithEncoder(cs, coder, val, CharacterCodingException.class);
     }
 
-    private static byte[] encodeWithEncoder(Charset cs, byte coder, byte[] val, boolean doReplace) {
+    private static <E extends Exception> byte[] encodeWithEncoder(
+            Charset cs, byte coder, byte[] val, Class<E> exceptionClass)
```

Comment:
Since `exceptionClass` is a dummy argument that can't actually control the type of exception thrown by this method, that it being null or not is the only thing that matters, I recommend documenting/commenting about this fact, as this is otherwise potentially confusing. Maybe a name to indicate this nature (I came up with `replacementSwitch` but this is for sure horrible) would help too I guess?